pl_bolts/datamodules/binary_emnist_datamodule.py

```diff
@@ -0,0 +1,180 @@
+            seed (int): Random seed to be used for train/val/test splits.
+            shuffle (bool): If ``True``, shuffles the train data every epoch.
+            pin_memory (bool): If ``True``, the data loader will copy Tensors into
+                               CUDA pinned memory before returning them.
+            drop_last (bool): If ``True``, drops the last incomplete batch.
+        """
+        if not _TORCHVISION_AVAILABLE:  # pragma: no cover
+            raise ModuleNotFoundError(
+                'You want to use EMNIST dataset loaded from `torchvision` which is not installed yet.'
+            )
+
+        super(BinaryEMNISTDataModule, self).__init__(  # type: ignore[misc]
+            data_dir=data_dir,
+            val_split=val_split,
+            num_workers=num_workers,
+            normalize=normalize,
+            batch_size=batch_size,
+            seed=seed,
+            shuffle=shuffle,
+            pin_memory=pin_memory,
+            drop_last=drop_last,
+            *args,
+            **kwargs,
+        )
+        self.split = split
+        self._check_and_update()
+
+    def _check_and_update(self):
+        if self._emnist_split_exists:
+            self._update_val_split()
+
+    def _update_val_split(self):
+        if (self.val_split is None):
+            if self.split_metadata.get('validation'):
+                self.val_split = int(self.split_metadata.get('num_test'))
+            else:
+                self.val_split = self._DEFAULT_NO_VALIDATION_VAL_SPLIT
+
+    @property
+    def _emnist_split_exists(self) -> bool:
+        """Checks if the split exists in emnist or not."""
+        if not any(s == self.split for s in self.dataset_cls.splits):
+            allowed_values = ', '.join([f'"{v}"' for v in self.dataset_cls.splits])
+            raise ValueError(
+                f'Invalid value provided for split (="{self.split}"). ' + f'Allowed splits are: {allowed_values}'
+            )
+        return True
+
+    @property
+    def split_metadata(self):
+        return self.dataset_cls._metadata.get('splits').get(self.split)
+
+    @property
+    def num_classes(self) -> int:
+        """
+        Return:
+           - for ``byclass``: 62
+           - for ``bymerge``: 47
+           - for ``balanced``: 47
+           - for ``digits``: 10
+           - for ``letters``: 47
+           - for ``mnist``: 10
+        """
+        # The _metadata is only added to EMNIST dataset
+        # to get split-specific metadata.
+        return self.split_metadata.get('num_classes')
+
+    def prepare_data(self, *args: Any, **kwargs: Any) -> None:
+        """
+        Saves files to data_dir
+        """
+
+        def _prepare_with_splits(split: str):
+            self.dataset_cls(self.data_dir, split=split, train=True, download=True)
+            self.dataset_cls(self.data_dir, split=split, train=False, download=True)
+
+        _prepare_with_splits(self.split)
```

Comment:
Why not making it simpler?
```suggestion
        self.dataset_cls(self.data_dir, split=split, train=True, download=True)
        self.dataset_cls(self.data_dir, split=split, train=False, download=True)
```

Reply:
@akihironitta Yes, sure. This completely makes sense. Thank you.